annnufan/assignment1/solution_1.cpp

```diff
@@ -0,0 +1,118 @@
+#include <iostream>
+#include <string>
+#include <cstring>
+#include <vector>
+#include <algorithm>
+#include <cctype>
+#include <functional>
+
+bool word_anagram(std::string word_1, std::string word_2, bool case_sens) {
+	if (!case_sens) {
+		std::transform(word_1.begin(), word_1.end(), word_1.begin(), ::tolower);
+		std::transform(word_2.begin(), word_2.end(), word_2.begin(), ::tolower);
+	}
+	if (word_1.length() != word_2.length()) {
+		return false;
+	}
+
+	std::vector<int> count(256, 0);
+	for (auto ch : word_1) {
+		count[ch]++;
+	}
+	for (auto ch : word_2) {
+		if (count[ch] > 0) {
+			count[ch]--;
+		} else {
+			return false;
+		}
+	}
+	return true;
+}
+
+void solve_for_word(bool case_sens) {

```

Comment:
Nice helper functions that handle all the I/O.
They are short, simple and easy to understand, easy to use whenever you want, and keep the logic separated so you can use the logic without the I/O. Good work!